### DISCUSSION TITLE
the project.pbxproj file in ios loses all its content after faslane lane has been run

### DISCUSSION THREAD

**Discussion by kigh143** [fastlane/fastlane#16905] [Q&A] [answered]
I  am new to Fastlane, I was trying to integrate Fastlane into my react native application(IOS). I am facing an issue where my **project.pbxproj** file in ios loses all its content after Faslane lane has been run. My git shows the changes made by Fastlane and the file is empty so have to discard the changes to attain my file contents. The first time I ran the lane beta it worked fine and updated testlFlight. 

**FastLane failing with this error**
error: No profiles for 'com.AppName' were found: Xcode couldn't find any iOS App Development provisioning profiles matching 'com.AppName'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild. (in target 'AppName' from project 'AppName')

**What I did** 
I have a react native application on testlFlight.
I initialised Fastlane into the ios folder of my project which created my AppFile and FastFile
Since I have certificates on my apple developers account, I am using Cert and Sigh to manage the code signing (I don't want to revoke my existing certificates).
I disabled automatic code signing in XCode.

**FastFile contents **
```
default_platform(:ios)

platform :ios do
  desc "Push a new beta build to TestFlight"
  lane :beta do
    increment_build_number(xcodeproj: "AppName.xcodeproj")
    update_code_signing_settings
    get_certificates
    get_provisioning_profile
    build_app(workspace: "AppName.xcworkspace", scheme: "AppName")
    upload_to_testflight
  end
end
```
Anyone has faced this before?

**Comment by kigh143** [comment]
The application builds fine on Xcode and it can also be archived for distribution to test flight.

**Comment by lyndsey-ferguson** [comment, marked as answer]
Do you know which of the actions is actually causing the problem? I would comment out each one except the first and then run the `beta` lane.

If that run doesn't delete the contents of the `project.pbxproj` file, uncomment the next line and repeat.

**Reply by lyndsey-ferguson** [reply]
None of those look inherently suspicious.

**Reply by kigh143** [reply]
@lyndsey-ferguson thank you, I tried removing but still found the contents deleted.

**Reply by lyndsey-ferguson** [reply]
You commented them all out and the file contents get deleted!? Then there is something else that is not in the lane that is the problem. Do you have a before_X or an after_X block (see this [article on lanes](https://docs.fastlane.tools/advanced/lanes/) )?

**Reply by kigh143** [reply]
I don't have these blocks, am fairly new to Fastlane. 
Let me read about this, Thank you.

**Reply by kigh143** [reply]
When using cert and sigh to handle code signing, What changes are supposed to made in  Xcode code signing configurations,

**Reply by lyndsey-ferguson** [reply]
Hello @kigh143 , it is not clear to me from your response if you've actually commented out each call. So, I'm going to provide some instructions for you to try:

Change your fastlane from this:

```ruby
platform :ios do
  desc "Push a new beta build to TestFlight"
  lane :beta do
    increment_build_number(xcodeproj: "AppName.xcodeproj")
    update_code_signing_settings
    get_certificates
    get_provisioning_profile
    build_app(workspace: "AppName.xcworkspace", scheme: "AppName")
    upload_to_testflight
  end
end
```
to
```ruby
platform :ios do
  desc "Push a new beta build to TestFlight"
  lane :beta do
    # increment_build_number(xcodeproj: "AppName.xcodeproj")
    # update_code_signing_settings
    # get_certificates
    # get_provisioning_profile
    # build_app(workspace: "AppName.xcworkspace", scheme: "AppName")
    # upload_to_testflight
  end
end
```
And run your fastlane. Do the contents of the `project.pbxproj` get removed? If not, change your Fastfile to this:

```ruby
platform :ios do
  desc "Push a new beta build to TestFlight"
  lane :beta do
    increment_build_number(xcodeproj: "AppName.xcodeproj")
    # update_code_signing_settings
    # get_certificates
    # get_provisioning_profile
    # build_app(workspace: "AppName.xcworkspace", scheme: "AppName")
    # upload_to_testflight
  end
end
```
And run your fastlane. Do the contents of the `project.pbxproj` get removed? If not, continue the pattern of uncommenting each call to determine where the problem is.

**Reply by kigh143** [reply]
thank you, let me go through the steps again.

**Reply by kigh143** [reply]
Hello, I was able it run through this process and the file is not deleting anymore. I don't know what had gone wrong. Thank you 
@lyndsey-ferguson.
